sdk/anomalydetector/ai-anomaly-detector-rest/src/isUnexpected.ts

```diff
@@ -144,52 +130,36 @@ function geParametrizedPathSuccess(method: string, path: string): string[] {
     const candidatePath = getPathFromMapKey(key);
     // Get each part of the url path
     const candidateParts = candidatePath.split("/");
-
-    // If the candidate and actual paths don't match in size
-    // we move on to the next candidate path
-    if (
-      candidateParts.length === pathParts.length &&
-      hasParametrizedPath(key)
-    ) {
-      // track if we have found a match to return the values found.
-      let found = true;
-      for (let i = 0; i < candidateParts.length; i++) {
-        if (
-          candidateParts[i]?.startsWith("{") &&
-          candidateParts[i]?.endsWith("}")
-        ) {
-          // If the current part of the candidate is a "template" part
-          // it is a match with the actual path part on hand
-          // skip as the parameterized part can match anything
-          continue;
-        }
-
-        // If the candidate part is not a template and
-        // the parts don't match mark the candidate as not found
-        // to move on with the next candidate path.
-        if (candidateParts[i] !== pathParts[i]) {
-          found = false;
-          break;
-        }
+    // track if we have found a match to return the values found.
+    let found = true;
+    for (let i = candidateParts.length - 1; i >= 0; i--) {
```

Comment:
This is the fix I have for parametrized hosts. Instead of checking from the front. Start at the end and walk our way back, if no mismatch at the end we assume it is the same path.

@MaryGao - What do you think? If this works correctly I think we should be able to easily port it to codegen.

@saintzou66 - would you mind verifying if this fix works?

Reply:
One concern for this is if we have similar suffix of two urls we may mimatch the correct one:

```
url1: http://localhost:80/service/v1.1/a/b/c      ==> swagger path section: a/b/c
url2: http://localhost:80/service/v1.1/a/b/a/b/c  ==> swagger path section: a/b/a/b/c
```

Then the `a/b/c` segment could match both of them. Actually the shorter one would be more accurate.

Reply:
Yeah, maybe we need a way to include the whole parametrized host to be sure the match is accurate

Reply:
Do we have such a scenario in this library? If not, we can go with this workaround while we do the long term fix in the codegen

Reply:
There is no this case in this library